cas-server-core-tickets/src/main/java/org/jasig/cas/ticket/registry/AbstractTicketRegistry.java

```diff
@@ -56,4 +62,71 @@ public int serviceTicketCount() {
                 this.getClass().getName(), Integer.MIN_VALUE);
       return Integer.MIN_VALUE;
     }
+
+    @Override
+    public final boolean deleteTicket(final String ticketId) {
+        if (ticketId == null) {
+            return false;
+        }
+
+        final Ticket ticket = getTicket(ticketId);
+        if (ticket == null) {
+            return false;
+        }
+
+        if (ticket instanceof TicketGrantingTicket) {
+            logger.debug("Removing children of ticket [{}] from the registry.", ticket.getId());
+            final TicketGrantingTicket tgt = (TicketGrantingTicket) ticket;
+            deleteChildren(tgt);
+
+            final Collection<ProxyGrantingTicket> proxyGrantingTickets = tgt.getProxyGrantingTickets();
+            for (final ProxyGrantingTicket proxyGrantingTicket : proxyGrantingTickets) {
+                logger.debug("Removing proxy-granting ticket [{}]", proxyGrantingTicket.getId());
+                deleteTicket(proxyGrantingTicket.getId());
+            }
+
+        }
+        logger.debug("Removing ticket [{}] from the registry.", ticket);
+        return deleteSingleTicket(ticketId);
```

Comment:
just a boolean valid/invalid (or deleted) field that can be updated in the reverse order before deleting them. That way the ticket is invalidated regardless of what happens or when with the delete actually happens.